src/differentiation/jaches_products.jl

```diff
@@ -207,10 +207,12 @@ struct FwdModeAutoDiffVecProd{F,U,C,V,V!} <: AbstractAutoDiffVecProd
 end
 
 function update_coefficients(L::FwdModeAutoDiffVecProd, u, p, t)
-    FwdModeAutoDiffVecProd(L.f, u, L.vecprod, L.vecprod!, L.cache)
+    f = update_coefficients(L.f, u, p, t)
```

Comment:
hmm, so you're actually going for solution quite similar to https://github.com/SciML/SciMLOperators.jl/issues/172, right? i.e. you're not changing the signature of `f`, that stays `f(u)`, right?

Reply:
for this soln to work, we need to add this line in `FunctionOperator's update_coefficients, right:
```
# Allow L.op to have u/p/t dependence (safe since update_coefficients! defaults to null-op)
update_coefficients!(L.op, u, p, t)
```
Otherwise the update coefficients for `FwdModeAutoDiffVecProd` etc. won't get called

Reply:
FwdModeAutoDiffVecProd is like a wrapper that manages the difference bw signatures. It's `mul` uses `auto_jacvec` that expects signature `f(u), f!(v, u)`

Reply:
FunctionOperator's update_coeffs does that already. pasting below

```julia
function update_coefficients!(L::FunctionOperator, u, p, t)
    ops = getops(L)
    for op in ops
        update_coefficients!(op, u, p, t)
    end     

    L.p = p
    L.t = t

    nothing
end
```

Reply:
ah, didn't realize that `op` was already part of `L`'s ops, great:)

Reply:
same for oop `update_coeffs`